File: src/dp/_718_find_length.rs
struct Solution;

impl Solution {
    pub fn find_length(nums1: Vec<i32>, nums2: Vec<i32>) -> i32 {
        let mut dp = vec![vec![0; nums2.len() + 1]; nums1.len() + 1];
        let mut max = 0_i32;
        for i in 1..=nums1.len() {
            for j in 1..=nums2.len() {
                if nums1[i - 1] == nums2[j - 1] {
                    let v = dp[i - 1][j - 1] + 1;
                    if v > max {
                        max = v;
                    }
                    dp[i][j] = v;
                }
            }
        }
        max
    }
}

File: leetcode-go/718_findLength.go
package leetcode_go

func findLength(nums1 []int, nums2 []int) int {
	ret := 0
	dp := make([][]int, len(nums1)+1)
	for i := range dp {
		dp[i] = make([]int, len(nums2)+1)
	}

	for i := 1; i < len(nums1)+1; i++ {
		for j := 1; j < len(nums2)+1; j++ {
			if nums1[i-1] == nums2[j-1] {
				dp[i][j] = dp[i-1][j-1] + 1
				if dp[i][j] > ret {
					ret = dp[i][j]
				}
			}
		}
	}
	return ret
}
